File: app-presentation/src/main/java/com/zxjdev/smile/presentation/user/settings/SettingsFragment.java
package com.zxjdev.smile.presentation.user.settings;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;

import com.zxjdev.smile.R;
import com.zxjdev.smile.presentation.common.SmileApplication;
import com.zxjdev.smile.presentation.common.base.fragment.BaseFragment;
import com.zxjdev.smile.presentation.infrastucture.splash.SplashActivity;

import javax.inject.Inject;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

public class SettingsFragment extends BaseFragment implements SettingsContract.View {

  public static final String TAG = SettingsFragment.class.getSimpleName();

  @BindView(R.id.btn_logout) Button btnLogout;

  @Inject SettingsPresenter settingsPresenter;

  public SettingsFragment() {
    setRetainInstance(true);
  }

  @Override
  public void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
  }

  @Nullable
  @Override
  public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
    View view = inflater.inflate(R.layout.fragment_settings, container, false);
    ButterKnife.bind(this, view);
    settingsPresenter.takeView(this);
    return view;
  }

  @Override
  public void onDestroy() {
    super.onDestroy();
    settingsPresenter.dropView();
  }

  @OnClick(R.id.btn_logout)
  public void logoutClick() {
    settingsPresenter.handleLogout();
  }

  @Override
  public void onLogoutSuccess() {
    ((SmileApplication) getActivity().getApplication()).releaseUserComponent();

    Intent intent = new Intent(getActivity(), SplashActivity.class);
    intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
    startActivity(intent);
  }
}

File: app-presentation/src/main/java/com/zxjdev/smile/presentation/user/settings/SettingsFragment.kt
package com.zxjdev.smile.presentation.user.settings

import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.zxjdev.smile.R
import com.zxjdev.smile.presentation.common.SmileApplication
import com.zxjdev.smile.presentation.common.base.fragment.BaseFragment
import com.zxjdev.smile.presentation.infrastucture.splash.SplashActivity
import kotlinx.android.synthetic.main.fragment_settings.*
import javax.inject.Inject

class SettingsFragment : BaseFragment(), SettingsContract.View {

    @Inject internal lateinit var settingsPresenter: SettingsPresenter

    init {
        retainInstance = true
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
            savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_settings, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        btn_logout.setOnClickListener { settingsPresenter.handleLogout() }
        settingsPresenter.takeView(this)
    }

    override fun onDestroy() {
        super.onDestroy()
        settingsPresenter.dropView()
    }

    override fun onLogoutSuccess() {
        (activity?.application as SmileApplication).releaseUserComponent()

        val intent = Intent(activity, SplashActivity::class.java)
        intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK
        startActivity(intent)
    }

    companion object {

        val TAG: String = SettingsFragment::class.java.simpleName
    }
}
